app/state_test.go

```diff
@@ -5405,3 +5409,53 @@ func TestFetchRetryUpdateControl(t *testing.T) {
 
 	assert.IsType(t, &fetchControlMapState{}, next)
 }
+
+func TestArtifactRollbackRebootUpdateModuleRebootHandling(t *testing.T) {
+	tempDir, _ := ioutil.TempDir("", "logs")
+	DeploymentLogger = NewDeploymentLogManager(tempDir)
+	defer func() {
+		DeploymentLogger = nil
+		os.RemoveAll(tempDir)
+	}()
+
+	tests := map[string]struct {
+		setInitialValue   bool
+		initialValue      datastore.RebootType
+		moduleRebootReply installer.RebootAction
+	}{
+		"Reboot already requested": {
+			setInitialValue:   true,
+			initialValue:      datastore.RebootTypeAutomatic,
+			moduleRebootReply: installer.RebootRequired,
+		},
```

Comment:
For completeness sake, maybe also test one where `moduleRebootReply: installer.NoReboot`? This is basically the old behavior, but doesn't hurt to verify that it still works.